### PR TITLE
Deadline: Allow disabling strict error check in Maya submissions

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
DL by default has Strict error checking, but some errors are not fatal. 

## Description
This allows to set profile based on Task and Subset values to temporarily disable Strict Error Checks.

Subset and task names should support regular expressions. (not wildcard notation though).

## Testing notes:
1. Configure Settings in `project_settings/deadline/publish/MayaSubmitDeadline/strict_error_checking` - set to false
3. create new render instance, publish from Maya to DL, check `MayaBatch Settings` of job properties
2023-02-02 18_42_10-Testing Platform PC - AnyDesk

### CODE REVIEW

File: openpype/modules/deadline/plugins/publish/submit_maya_deadline.py [hunk 1]
```diff
@@ -64,6 +65,7 @@ class MayaPluginInfo(object):
     # Include all lights flag
     RenderSetupIncludeLights = attr.ib(
         default="1", validator=_validate_deadline_bool_value)
+    StrictErrorChecking = attr.ib(default="1")
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/plugins/publish/submit_maya_deadline.py:68]: Should it be string or int?
```suggestion
    StrictErrorChecking = attr.ib(default=1)
```

File: openpype/settings/entities/schemas/projects_schema/schema_project_deadline.json [hunk 2]
```diff
@@ -195,6 +195,36 @@
...
+                        },
+                        {
+                            "type": "list",
+                            "collapsible": true,
+                            "key": "disable_strict_check_profiles",
+                            "label": "Disable Strict Error Check profiles",
+                            "use_label_wrap": true,
+                            "docstring": "Set profile for disabling 'Strict Error Checking'",
+                            "object_type": {
+                                "type": "dict",
```
Comment (iLLiCiTiT) [inline openpype/settings/entities/schemas/projects_schema/schema_project_deadline.json:207]: I would add one boolean input which would tell if it should be strict or not. There is no way how to tell: By default enable strict error check, except one very specific case.

Comment (iLLiCiTiT) [reply openpype/settings/entities/schemas/projects_schema/schema_project_deadline.json:207]: Or maybe for all `Scene` task types make it strict, but if task name is `TestScene` then ignore it.

Comment (iLLiCiTiT) [reply openpype/settings/entities/schemas/projects_schema/schema_project_deadline.json:207]: Maybe: In that case the logic should be reversed to "enable" strict check profiles, so the profile would enable strict mode or disable strict mode?

Comment (kalisp) [reply openpype/settings/entities/schemas/projects_schema/schema_project_deadline.json:207]: Default is Strict check enabled, that is DL default, even when we didn't sending anything.
This functionality is only as a workaround to push trough some critical publish which is failing on minor recoverable error.

We are now discussing with @m-u-r-p-h-y  if profiles are necessary, if simple boolean toggle wouldn't be enough.

**Review by m-u-r-p-h-y — CHANGES_REQUESTED** [review]
I'm afraid we are overcomplicating this. Sorry for the confusion. I would suggest simplifying the whole process and creating simple default settings in 

`project_settings/deadline/publish/MayaSubmitDeadline` for Strict Error Checking (default on)

strictError

and reading the value when creating a render instance in Maya. This way user can decide to switch it off for particular scene which is failing on Deadline farm for a non fatal reason.

strictErrorMaya

**Comment by kalisp** [comment]
Removed profiles, changed to boolean, set to render instance.

File: openpype/modules/deadline/plugins/publish/submit_maya_deadline.py [hunk 3]
```diff
@@ -104,6 +105,7 @@ class MayaSubmitDeadline(abstract_submit_deadline.AbstractSubmitDeadline):
     jobInfo = {}
     pluginInfo = {}
     group = "none"
+    disable_strict_check_profiles = []
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/plugins/publish/submit_maya_deadline.py:108]:
```suggestion
```

**Review by iLLiCiTiT — APPROVED** [review]
One small suggestion, otherwise approved.

**Review by m-u-r-p-h-y — CHANGES_REQUESTED** [review]
it does not work me.

Maya2022
Deadline 10.2.0.9

newly created Render instance is picking correct value from settings (false), but submitted job has StrictErrorChecking enabled anyway

image

**Comment by kalisp** [comment]
Fixed propagation.

**Review by m-u-r-p-h-y — APPROVED** [review]
now everything works as expected.

tested a couple of scenarios.

Settings = true
Render instance = false

Settings = true
Render instance = true

Settings = false
Render instance = true

**Pull request merged**